### ISSUE TITLE
Compile errors on geany-1.37.1 and geany-plugins-1.37

### ISSUE THREAD

**Issue by slackuser0xae34** [geany/geany#2792]
# Problems with compiling Geany 1.37.1 on Slackware 14.2:

First Problem:

$ /bin/sh .../geany-1.37.1/build-aux/install-sh -d .../package-geany/usr/share/geany

/bin/ginstall -c -m 644 ./COPYING /home/ric/TMP/package-geany/usr/share/geany/GPL-2
 /bin/mkdir -p '/home/ric/TMP/package-geany/usr/share/applications'
 /bin/ginstall -c -m 644 geany.desktop '/home/ric/TMP/package-geany/usr/share/applications'
 /bin/mkdir -p '/home/ric/TMP/package-geany/usr/lib/pkgconfig'
 /bin/ginstall -c -m 644 geany.pc '/home/ric/TMP/package-geany/usr/lib/pkgconfig'
make[2]: Leaving directory '/home/ric/TMP/geany-1.37.1'
make[1]: Leaving directory '/home/ric/TMP/geany-1.37.1'
cp: cannot stat 'COMMITTERS': No such file or directory

The package geany-1.37.1.tar.gz is missing the "COMMITTERS" file, as such compilation fails. Copying the "COMMITTERS" file from the 1.34.1 package to the 1.37.1 package, then recompiling works on a 32 bit laptop.

Second Problem:

On a 64 bit desktop, compiling the 1.37.1 packages results in the following error message: 

/usr/lib64/gcc/x86_64-slackware-linux/5.5.0/../../../../x86_64-slackware-linux/bin/ld: i386 architecture of input file `../scintilla/.libs/libscintilla.a(DBCS.o)' is incompatible with i386:x86-64 output

collect2: error: ld returned 1 exit status
Makefile:708: recipe for target 'libgeany.la' failed
make[3]: *** [libgeany.la] Error 1

The issue is in ../scintilla/.libs/libscintilla.a(DBCS.o):

$ file DBCS.o 
DBCS.o: ELF 32-bit LSB relocatable, Intel 80386, version 1 (SYSV), not stripped

Running a "make clean" before compiling is desirable.

After correcting successful compiling: 
$ diff -q geany-1.37.1.working/ geany-1.37.1/ > geany.diff

Only in geany-1.37.1.working/: COMMITTERS

PROBLEMS WITH COMPILING Geany-plugins 1.37:

First Problem:

There are "missing files in the '.gz file (or the makefile needs tweaking).

diff -q geany-plugins-1.37.working/ geany-plugins-1.37/ > geany-plugins.diff

Only in geany-plugins-1.37.working/: HACKING
Only in geany-plugins-1.37.working/: MAINTAINERS
Only in geany-plugins-1.37.working/: autogen.sh

Copying the missing files from the 1.34.1 package to the 1.37.1 package, then recompiling works.

Second Problem:

 In installing geany/geany-plugins I was also experimenting with updating gpg & gpg2 (from Slackware-current (aka Slackware 15-rc1). I need to update libassuan from version 2.4.2 to version 2.5.5. There is an issue with geany-plugins and libassuan 2.5.5. One of the libraries which geany-plugins depends on is no longer present in libassuan 2.5.5. 
     After reverting back to the default gpg/gpg2/libassuan (libassuan 2.4.2) the problem disappeared. Eventually geany-plugins may need a tweak for later version of libassuan.

Help: does anyone know how to shut off the github markdown stuff and use just plain old text? Thanks. S.

**Comment by elextr** [comment]
Why are you not just following the instructions in INSTALL `make install` instead of miscellaneous scripts in `build-aux`?

**Comment by elextr** [comment]
If you are using a tarfile from slackware (not the tarball from the Geany website) please send the issues there, Geany does not make those.

**Comment by slackuser0xae34** [comment]
I am using the tarball from the Geany website (https://download.geany.org/geany-1.37.1.tar.gz and https://plugins.geany.org/geany-plugins/geany-plugins-1.37.tar.gz). I adjusted the slackware build script (from slackbuilds.org) from Geany 1.34 to 1.37.  

When the slackbuild script failed to compile correctly, I unpacked the gz files and looked into the build process. Manually building according to the instructions still failed. I troubleshot the problem and what I found is listed above.

Thanks for the reply. S.

**Comment by elextr** [comment]
Please try making a _new_ clean unused untar of the Geany tarball somewhere say `/home/you/geany_tarball`, 

```
cd /home/you/geany_tarball/geany
configure --prefix=/home/you/geany_tarball
make install
cd ../bin
./geany -c ../config
```

That keeps everything in the `geany_tarball` directory so you can wipe it easily at any time.  If there are failures to build or run using that method paste the errors here.

Otherwise, as I said above we don't do slackware build scripts etc so can't help when anything you do with them fails because we know nothing about them.

**Comment by slackuser0xae34** [comment]
I downloaded a fresh copy from the Geany download site and tried your instructions as above and they worked as you would expect. Geany runs successfully from the local directory. 

Why it didn't work before I don't know. I regularly check and double check my work then sleep on the problem before raising an issue publicly. Perhaps my efforts I still made an error. I'll accept that things are working as they should.

FTR, I included the stuff regarding Slackware and slackbuilds scripts as background information. I have often found that background information helps in diagnosing problems, though, admittedly sometimes background info clutters up the description. I didn't expect anyone here to help diagnosing slackware/slackbuild problems.

Thanks S.
